src/backend.ts

```diff
@@ -311,14 +321,52 @@ export function setupSuspendResumeHandler(): () => void {
 export function setupFocusChangeHandler(): () => void {
   let appIsStartingUp: boolean = false;
   let lastPid = 0;
+  let lastAppid = 0;
+  let validKeyEvent: SystemKeyEvent | null = null;
+  SteamClient.System.UI.RegisterForSystemKeyEvents((e) => {
+    // Have not found any race condition issues with this approach since the key event fires long before focus change
+    validKeyEvent = e;
+  });
+
   const { unregister: unregisterFocusChangeEvents } =
     SteamClient.System.UI.RegisterForFocusChangeEvents(
       throttle(async (fce: FocusChangeEvent) => {
         // don't try anything while an application is launching or it could pause it midlaunch
         if (appIsStartingUp) return;
         // skip if we already got such an event before
-        if (fce.focusedApp.pid === lastPid) return;
+        if (
+          fce.focusedApp.pid === lastPid &&
+          fce.focusedApp.appid === lastAppid
+        )
+          return;
         lastPid = fce.focusedApp.pid;
+        lastAppid = fce.focusedApp.appid;
+        // Only pause if the user has enabled it
+        if (
+          fce.focusedApp.appid === 769 &&
+          (await loadSettings()).overlayPause &&
```

Comment:
I think it would be preferable to have the value dependent on the promise as last test. It doesn't make a lot of difference in the grand scheme of things but I would put simple tests (number and string) before more complex ones so that there's an opportunity to escape the condition faster :)

Reply:
Added!